### PR TITLE
Add `TableColumnReferenceToken` to know when a table is referenced by a column

### PR DESCRIPTION
**Pull request by KathiaBarahona**
Created TableColumnReferenceToken, this way we can know the table being references by a column.

### CODE REVIEW

File: bach/bach/expression.py [hunk 1]
```diff
@@ -282,18 +299,49 @@ def has_windowed_aggregate_function(self) -> bool:
...
-                result.append(data_item.resolve(dialect, table_name))
+                result.append(data_item.resolve(table_name))
             else:
                 result.append(data_item)
         return self.__class__(result)
 
+    def remove_table_column_references(self) -> Tuple[str, str, 'Expression']:
+        """
+        removes all table references from this expression.
+        Returns first table_name and column_name found and a new expression without table column references
```
Comment (jansentom) [inline bach/bach/expression.py:326]: Should it not return as list of (table_name, column_name) tuples? The first one is quite random :)

**Review by jansentom — APPROVED** [review]
One minor comment, rest LGTM

File: bach/bach/expression.py (hunk 1 above)
Comment (KathiaBarahona) [reply bach/bach/expression.py:326]: I thought about returning a mapping between table_names and list of columns, buuuuuut make things more complicated and in theory we shouldn't be allowing multiple table refs in an expression.

Comment (KathiaBarahona) [reply bach/bach/expression.py:326]: also, usually the first column found in the expression is the name of the series. It's "hacky" but prefer to call it an heuristic ;)

Comment (jansentom) [reply bach/bach/expression.py:326]: is the "in theory" guarded aka do we raise when that happens? Maybe we should check in this method that it's actually u
unique or raise, that'd be nice?

Comment (KathiaBarahona) [reply bach/bach/expression.py:326]: nope, :thinking: the only place where we set the table alias is in "merge" and it's hardcoded. Not sure, in that case probably we will need to "preprocess" the expressions in the SqlModel

Comment (KathiaBarahona) [reply bach/bach/expression.py:326]: but will raise it here :P

Comment (thijs-obj) [reply bach/bach/expression.py:325]: I don't think it makes much sense to raise the exception only here. This is an okay second place to do it. But I think that if we say that `"we shouldn't be allowing multiple table refs in an expression"`, then we should raise that exception when we create such an illegal `Expression`.

Comment (thijs-obj) [reply bach/bach/expression.py:325]: Also, in a merge could we not have an expression that references two different tables, or am I misunderstanding that?

Comment (thijs-obj) [reply bach/bach/expression.py:325]: I do agree that returning a list or map of table-names wouldn't be very useful for code that calls this probably.

**Pull request merged**